cmd/storage-rest-server.go

```diff
@@ -52,7 +53,18 @@ var errDiskStale = errors.New("drive stale")
 
 // To abstract a disk over network.
 type storageRESTServer struct {
-	storage *xlStorageDiskIDCheck
+	storage atomic.Value
+}
+
+func (s *storageRESTServer) getStorage() *xlStorageDiskIDCheck {
+	if s, ok := s.storage.Load().(*xlStorageDiskIDCheck); ok {
+		return s
+	}
+	return nil
+}
```

Comment:
so this has the potential to return `nil` so a crash is okay here? @vadmeste

Reply:
This still applies @vadmeste

Reply:
@harshavardhana the caller function will be responsible for not crashing

Reply:
so `s.getStorage().WalkDir()` will crash?

Reply:
> so s.getStorage().WalkDir() will crash?

what do you mean ?

Reply:
That these will crash if it returns nil: `infos, err := s.getStorage().ListVols(r.Context())`

Reply:
yes, it will crash